### PR TITLE
[fixes #24] support PNP and basedir slashes

### LINKED ISSUE #24: PNP and basedir trailing slashes
When calling PNP's `resolveToUnqualified` method, when the issuer represents a directory the issuer must have a trailing slash.

As far as I can tell, resolve-package-path deals only in base directories, never base *files*, so I think it should always append a trailing slash to basedir before calling `resolveToUnqualified`.

Here are reproduction steps:

1. Clone this repo.
2. cd `resolve-package-path`
3. `volta pin node@13` 
4. `volta pin yarn@1.22.4`
5. `yarn set version berry`
6. `yarn`
7. Start interactive node with `yarn node`
8. Evaluate these steps:

    ```js
    // load our requirePackagePath function
    rpp = require('.')

    // use it to locate chai. This one happens to work even without a trailing slash, I think 
    // because the issuer is a real path on the filesystem
    chaiDir = path.dirname(rpp('chai', '.'))
    // Result: '/Users/edward/hacking/resolve-package-path/.yarn/cache/chai-npm-4.2.0-b777309e5a-2.zip/node_modules/chai'

    // this is a valid dependency of chai, but it returns null! In this case, the issuer is a hypothetical
    // directory inside a zip file that only yarn pnp understands. I suspect that is why the
    // trailing slash is mandatory.
    rpp('deep-eql', chaiDir)
    // Result: null

    // you can make it get the right answer like this:
    rpp('deep-eql', chaiDir + '/')
    // Result: '/Users/edward/hacking/resolve-package-path/.yarn/cache/deep-eql-npm-3.0.1-9a66c09c65-2.zip/node_modules/deep-eql/package.json'
    ```

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

File: tests/index-test.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ describe('resolve-package-path', function () {
   });
 
   if (require('os').platform() !== 'win32') {
-    describe.only('yarn pnp usage', function () {
```
Comment (rwjblue) [inline tests/index-test.ts]: 😱 waaaat, we had this `.only` landed in master? 😭

Comment (stefanpenner) [reply tests/index-test.ts]: ooops!!

**Review by rwjblue — APPROVED** [review]

File: tests/index-test.ts (hunk 1 above)
Comment (rwjblue) [reply tests/index-test.ts]: ya was introduced by https://github.com/stefanpenner/resolve-package-path/commit/448d3a19f9bc5b8b0855f0c50950a0e7937eb4fb

**Review by ef4 — APPROVED** [review]

**Comment by stefanpenner** [comment]
released as v4.0.1 🎉

**Pull request merged**